lib/entities/semantic-file.js

```diff
@@ -0,0 +1,161 @@
+import { SERVICE_URI } from '../../env';
+import { filenameToUri, uriToFilename } from '../util/file';
+import { v5 as uuidv5 } from 'uuid';
+import path from 'path';
+import fs from 'fs-extra';
+import Base64 from 'crypto-js/enc-base64';
+import sha256 from 'crypto-js/sha256';
+import moment from 'moment';
+
+const NAMESPACE = '77219b1e-e23d-4828-847f-55e2d7fac687'; // NOTE: used for consistent uuid generation based on the URI
+
+/**
+ * Class representation of a semantic-file
+ */
+export class SemanticFile {
+
+  static exists(filename) {
+    if (filename instanceof SemanticFile) {
+      filename = filename.filename;
+    }
+    return fs.existsSync(filename);
+  }
+
+  static write(filename, content) {
+    if (filename instanceof SemanticFile) {
+      filename = filename.filename;
+    }
+    return fs.writeFileSync(filename, content);
+  }
+
+  static contentHash(content) {
+    if (content instanceof SemanticFile) {
+      content = content.content;
+    }
+    return Base64.stringify(sha256(content));
+  }
+
+  constructor({uri, filename, created = moment(), modified = moment(), publisher = SERVICE_URI}) {
+    this.physicalURI = uri ? uri : filenameToUri(filename);
+    this.created = created;
+    this.modified = modified;
+    this.publisher = publisher;
+  }
+
+  get uri() {
+    return this.physicalURI;
+  }
+
+  get uuid() {
+    return uuidv5(this.physicalURI, NAMESPACE);
+  }
+
+  get virtualURI() {
+    return `http://data.lblod.info/files/${this.uuid}`;
+  }
+
+  get filename() {
+    return uriToFilename(this.physicalURI);
+  }
+
+  get extension() {
+    return path.extname(this.filename);
+  }
+
+  /**
+   * TODO:  this is hardcode for now, might consider a plugin later
+   *        if media-types really start going beyond what is expected now.
+   *
+   * @returns {string}
+   */
+  get format() {
+    switch (this.extension) {
+      case '.json':
+        return 'application/json';
+      case '.ttl':
+        return 'application/n-triples';
+      case '.js':
+        return 'application/javascript';
+      default:
+        return 'plain/text';
+    }
+  }
+
+  get content() {
+    switch (this.extension) {
+      case '.json':
+        return require(this.filename);
+      case '.js':
+        return require(this.filename);
+      default:
+        return fs.readFileSync(this.filename, 'utf8');
+    }
+  }
+
+  toNT() {
+    let buffer = [];
+    const prefixes = [
+      '@prefix nfo: <http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#> .',
+      '@prefix nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#> .',
+      '@prefix mu: <http://mu.semte.ch/vocabularies/core/> .',
+      '@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .',
+      '@prefix dbpedia: <http://dbpedia.org/resource/> .',
+      '@prefix dct: <http://purl.org/dc/terms/> .',
+    ];
+
+    const escaped = {
+      virtualURI: `<${this.virtualURI}>`,
+      physicalURI: `<${this.physicalURI}>`,
```

Comment:
Woulnd't `sparqlEscape` work there too ? It would fall under the [sparqlEscapeUri](https://github.com/mu-semtech/mu-javascript-template/blob/master/helpers/mu/sparql.js#L134) case no ?

Reply:
Yes, that's a fair observation. I think I just started copy pasting it as such but I'll refactor it!